protocol/kafka_confluent/v2/protocol.go

```diff
@@ -0,0 +1,225 @@
+/*
+ Copyright 2023 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package kafka_confluent
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+
+	cecontext "github.com/cloudevents/sdk-go/v2/context"
+)
+
+var (
+	_ protocol.Sender   = (*Protocol)(nil)
+	_ protocol.Opener   = (*Protocol)(nil)
+	_ protocol.Receiver = (*Protocol)(nil)
+	_ protocol.Closer   = (*Protocol)(nil)
+)
+
+type Protocol struct {
+	kafkaConfigMap *kafka.ConfigMap
+
+	consumer             *kafka.Consumer
+	consumerTopics       []string
+	consumerRebalanceCb  kafka.RebalanceCb     // optional
+	consumerPollTimeout  int                   // optional
+	consumerErrorHandler func(err kafka.Error) //optional
+	consumerMux          sync.Mutex
+
+	producer             *kafka.Producer
+	producerDeliveryChan chan kafka.Event // optional
+	producerDefaultTopic string           // optional
+
+	// receiver
+	incoming chan *kafka.Message
+}
+
+func New(opts ...Option) (*Protocol, error) {
+	p := &Protocol{
+		consumerPollTimeout: 100,
+		incoming:            make(chan *kafka.Message),
+	}
+	if err := p.applyOptions(opts...); err != nil {
+		return nil, err
+	}
+
+	if p.kafkaConfigMap != nil {
+		if p.consumerTopics != nil && p.consumer == nil {
+			consumer, err := kafka.NewConsumer(p.kafkaConfigMap)
+			if err != nil {
+				return nil, err
+			}
+			p.consumer = consumer
+		}
+		if p.producerDefaultTopic != "" && p.producer == nil {
+			producer, err := kafka.NewProducer(p.kafkaConfigMap)
+			if err != nil {
+				return nil, err
+			}
+			p.producer = producer
+		}
+		if p.producer == nil && p.consumer == nil {
+			return nil, fmt.Errorf("at least receiver or sender topic must be set")
```

Comment:
sorry, just saw: any linter would flag this at some point: `Errorf` without variable placeholders (same with other errors)

Reply:
Done.